spec/reek/context/method_context_spec.rb

```diff
@@ -1,94 +1,93 @@
-      mc.record_call_to s(:send, s(:ivar, :@cow), :feed_to)
-      expect(mc.envious_receivers).to be_empty
+      method_context.record_call_to s(:send, s(:ivar, :@cow), :feed_to)
+      expect(method_context.envious_receivers).to be_empty
     end
 
     it 'should ignore explicit calls to self' do
-      mc.refs.record_reference_to :other
-      mc.record_call_to s(:send, s(:self), :thing)
-      expect(mc.envious_receivers).to be_empty
+      method_context.refs.record_reference_to :other
+      method_context.record_call_to s(:send, s(:self), :thing)
+      expect(method_context.envious_receivers).to be_empty
     end
 
     it 'should ignore implicit calls to self' do
-      mc.record_call_to s(:send, s(:lvar, :text), :each, s(:arglist))
-      mc.record_call_to s(:send, nil, :shelve, s(:arglist))
-      expect(mc.envious_receivers).to be_empty
+      method_context.record_call_to s(:send, s(:lvar, :text), :each, s(:arglist))
+      method_context.record_call_to s(:send, nil, :shelve, s(:arglist))
+      expect(method_context.envious_receivers).to be_empty
     end
 
     it 'should record envious calls' do
-      mc.record_call_to s(:send, s(:lvar, :bar), :baz)
-      expect(mc.envious_receivers).to include(:bar)
+      method_context.record_call_to s(:send, s(:lvar, :bar), :baz)
+      expect(method_context.envious_receivers).to include(:bar)
     end
   end
-end
-
-RSpec.describe Reek::Context::MethodParameters, 'default assignments' do
-  def assignments_from(src)
-    exp = Reek::Source::SourceCode.from(src).syntax_tree
-    ctx = Reek::Context::MethodContext.new(Reek::Context::RootContext.new, exp)
-    ctx.parameters.default_assignments
-  end
 
-  context 'with no defaults' do
-    it 'returns an empty hash' do
-      src = 'def meth(arga, argb, &blk) end'
-      expect(assignments_from(src)).to be_empty
+  describe '#default_assignments' do
+    def assignments_from(src)
+      exp = Reek::Source::SourceCode.from(src).syntax_tree
+      ctx = Reek::Context::MethodContext.new(nil, exp)
+      # FIXME: Create delegating method on MethodContext
+      ctx.default_assignments
     end
-  end
 
-  context 'with 1 default' do
-    before :each do
-      src = 'def meth(arga, argb=456, &blk) end'
-      @defaults = assignments_from(src)
-    end
-    it 'returns the param-value pair' do
-      expect(@defaults[0]).to eq [:argb, s(:int, 456)]
+    context 'with no defaults' do
+      it 'returns an empty hash' do
+        src = 'def meth(arga, argb, &blk) end'
+        expect(assignments_from(src)).to be_empty
+      end
     end
-    it 'returns the nothing else' do
-      expect(@defaults.length).to eq(1)
-    end
-  end
 
-  context 'with 2 defaults' do
-    before :each do
-      src = 'def meth(arga=123, argb=456, &blk) end'
-      @defaults = assignments_from(src)
+    context 'with 1 default' do
+      before :each do
+        src = 'def meth(arga, argb=456, &blk) end'
+        @defaults = assignments_from(src)
+      end
```

Comment:
P.S.: I know it's legacy code but it would be a quick fix and we'd still follow the boyscout rule.:)